sdk/servicebus/service-bus/src/core/streamingReceiver.ts

```diff
@@ -294,7 +294,22 @@ export class StreamingReceiver extends MessageReceiver {
         }
         return;
       } finally {
-        this._receiverHelper.addCredit(1);
+        if (this.receiveMode === "receiveAndDelete" || numberOfEmptyIncomingSlots(this.link) > 1) {
+          this._receiverHelper.addCredit(1);
+        } else {
+          // Additionally.. have a checkWithTimeout that keeps checking if the above if-condition satisfies
+          // If it ever satisfies - add the credit
+          this._onError?.({
+            error: new ServiceBusError(
+              `Circular buffer that contains the incoming deliveries is full, please settle the messages using settlement methods such as .completeMessage() on the receiver.
+                Or set the "autoComplete" flag to true to let the library complete the messages on your behalf.`,
+              "GeneralError"
```

Comment:
Is this marked as retryable?

Reply:
`retryable` is true by default, so fine?